### ISSUE TITLE
Panic on line 41

### ISSUE THREAD

**Issue by danepowell** [jerrywoo96/AddMissingHeaders#2]
Whenever a middleware using this plugin runs, it generates a panic in the Traefik logs:

> time="2022-12-30T16:01:39Z" level=error msg="plugins-storage/sources/gop-3601781398/src/github.com/jerrywoo96/AddMissingHeaders/addmissingheaders.go:41:26: panic" module=github.com/jerrywoo96/AddMissingHeaders plugin=plugin-AddMissingHeaders

There are some debug logs around that but I'm hesitant to post them as they have some sensitive info. Let me know if you need them.

Here's my middleware config:

```
    append-security-headers:
      plugin:
        AddMissingHeaders:
          responseHeaders:
            # headers: frameDeny can break apps that use frames such as pi-hole
            # @see https://github.com/traefik/traefik/issues/8808
            X-Frame-Options: DENY
```

I thought maybe the problem was that I hadn't defined any requestHeaders so I added the following, but it didn't help:
```
          requestHeaders:
            X-Custom-RequestHeader: CustomRequestHeader
```

**Comment by jerrywoo96** [comment]
Now that you mentioned it, i checked my traefik logs, there are other plugins created by other people that also produces this panic error, not just mine. So i guess its a problem with traefik's yaegi.

![Screenshot_2023-01-01-08-59-09-876-edit_com.lonelycatgames.Xplore.jpg](https://user-images.githubusercontent.com/26324139/210158479-90a0cb8d-8184-4c30-b777-01685426076e.jpg)

**Comment by jerrywoo96** [comment]
> Whenever a middleware using this plugin runs, it generates a panic in the Traefik logs:
> 
> > time="2022-12-30T16:01:39Z" level=error msg="plugins-storage/sources/gop-3601781398/src/github.com/jerrywoo96/AddMissingHeaders/addmissingheaders.go:41:26: panic" module=github.com/jerrywoo96/AddMissingHeaders plugin=plugin-AddMissingHeaders
> 
> There are some debug logs around that but I'm hesitant to post them as they have some sensitive info. Let me know if you need them.
> 
> Here's my middleware config:
> 
> ```
>     append-security-headers:
>       plugin:
>         AddMissingHeaders:
>           responseHeaders:
>             # headers: frameDeny can break apps that use frames such as pi-hole
>             # @see https://github.com/traefik/traefik/issues/8808
>             X-Frame-Options: DENY
> ```
> 
> I thought maybe the problem was that I hadn't defined any requestHeaders so I added the following, but it didn't help:
> 
> ```
>           requestHeaders:
>             X-Custom-RequestHeader: CustomRequestHeader
> ```

https://github.com/traefik/traefik/issues/8937

**Comment by danepowell** [comment]
Thanks, seems like an upstream issue with Traefik, and there's even an open PR to fix it 👀

**Comment by jerrywoo96** [comment]
@danepowell 

Yaegi v0.15.0 has been implemented into Traefik v2.9.7.
See if you still encounter any panic logs.

**Comment by danepowell** [comment]
Updated to 2.9.7, so far so good! Thanks.
